DependencyInjection/Configuration.php

```diff
@@ -99,7 +99,21 @@ private function getMailersNode()
                     return $v;
                 })
             ->end()
+            // Deprecate url
+            ->beforeNormalization()
+                ->ifTrue(function ($v) {
+                    return is_array($v) && array_key_exists('url', $v);
```

Comment:
I will keep this condition to keep the `trigger_error` anyway, but check if `dsn` is already set.